src/canvas/Canvas.ts

```diff
@@ -1546,13 +1542,10 @@ export class Canvas extends SelectableCanvas implements CanvasOptions {
       { includeIntersecting: !this.selectionFullyContained }
     ) as FabricObject[];
 
-    const objects = isClick
-      ? collectedObjects[0]
-        ? [collectedObjects[0]]
-        : []
-      : collectedObjects.length > 1
-      ? collectedObjects.filter((object) => !object.onSelect({ e })).reverse()
-      : collectedObjects;
+    const objects =
+      collectedObjects.length > 1
+        ? collectedObjects.filter((object) => !object.onSelect({ e })).reverse()
+        : collectedObjects;
```

Comment:
why don't we call onSelect in this case?

Reply:
because it gets solved by setActiveObject shortly below, while the object.length > 1 uses an collection.add method

Reply:
I see
This requires a comment